### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by acaliaro** [xamarin/XamarinCommunityToolkit#984]
### Description
I would like to know the width and height of a StackLayout when is rendered.
I know there is `xct:LifecycleEffect` that could help but I don't understand how it works

I use 

  ```
      <StackLayout
            BackgroundColor="Yellow"
            Style="{StaticResource PopupExternalStackLayoutStyle}"
            VerticalOptions="Center">
            <StackLayout.Effects>
                <xct:LifecycleEffect Loaded="LifecycleEffect_Loaded" Unloaded="LifecycleEffect_Unloaded"/>
            </StackLayout.Effects>

...
...
```

and

```
 private void LifecycleEffect_Loaded(object sender, System.EventArgs e)
        {
            System.Diagnostics.Debug.WriteLine("loaded width=" + ((StackLayout)sender).Width + " height=" + ((StackLayout)sender).Height);
        }

        private void LifecycleEffect_Unloaded(object sender, System.EventArgs e)
        {
            System.Diagnostics.Debug.WriteLine("unloaded width=" + ((StackLayout)sender).Width + " height=" + ((StackLayout)sender).Height);

        }
```

But I have these logs:

[0:] loaded width=-1 height=-1
[0:] unloaded width=332 height=142

So, I receive the Width and height when the layout is unloaded. Is it possible to know its widht and height when it  is loaded?
### Steps to Reproduce

1. Create a StackLayout and add xct:LifecycleEffect 
2. Add Loaded and Unloaded events
3. take a look to its dimension

### Expected Behavior
Have StackLayout's Width and Height in loaded event
### Actual Behavior
always -1

### Basic Information

- Version with issue: 1.1.0-preview1021
- Last known good version: unknown
- IDE: Visual Studio 2019 16.8.5
- Platform Target Frameworks: 
  - iOS:  
  - Android:  10
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

**Comment by pictos** [comment]
@acaliaro this may be because of the way how XF handles controls. The loaded event tells you when the controls are loaded on memory, and probably, the StackLayout didn't measure itself. I suggest you listen to the `SizeChanged` event.

**Comment by acaliaro** [comment]
> @acaliaro this may be because of the way how XF handles controls. The loaded event tells you when the controls are loaded on memory, and probably, the StackLayout didn't measure itself. I suggest you listen to the `SizeChanged` event.

Yes @pictos , with `SizeChanged` it works, but in your opinion it shouldn't work with LifecycleEffect too? When the "Loaded" event arrives the layout should be fully rendered and therefore the dimensions should also be correct

**Comment by pictos** [comment]
@acaliaro that depends, for all platforms but iOS I use the native events to know when the control/container is allocated on memory. We have the Forms layer that does calculations and initialization of the renderer, so there's a time between the control is allocated and the control is measured.
